mesonbuild/backend/xcodebackend.py

```diff
@@ -1229,6 +1229,13 @@ def generate_custom_target_shell_build_phases(self, objects_dict):
             if not isinstance(t, build.CustomTarget):
                 continue
             (srcs, ofilenames, cmd) = self.eval_custom_target_command(t, absolute_outputs=True)
+            # Fix object paths, as they point to non-existent private directories.
+            for i, _ in enumerate(srcs):
+                if srcs[i].endswith(".o"):
+                    srcs[i] = '/'.join(srcs[i].split('/')[2:])
+            for i, _ in enumerate(cmd):
+                if cmd[i].endswith(".o"):
+                    cmd[i] = '/'.join(cmd[i].split('/')[2:])
```

Comment:
You're not using the enumeration here, the generally superior approach would be `range(len(srcs))`...

... but also why not use the enumeration directly?

```python
            for i, src in enumerate(srcs):
                if src.endswith(".o"):
                    srcs[i] = '/'.join(src.split('/')[2:])
```

Reply:
I'm not certain I understand what's going on here, by the way. Isn't this supposed to be handled by the xcode specific `def get_target_dir`?

Reply:
I do have another idea: Modify `_determine_ext_objs()` in `backends.py` so that it skips adding our target private directory to object paths when the Xcode backend is in use. That way, enumerations can be avoided and we won't have to fix object paths after they are generated.

Reply:
You probably want to override the method for xcodebackend.py, but yes.

Reply:
Most of the logic in that method would need to be copied over in that case. My approach would be to avoid that and check the name of the backend:

```diff
-            objpath = os.path.join(proj_dir_to_build_root, targetdir, objname)
+            if self.name != 'xcode':
+                objpath = os.path.join(proj_dir_to_build_root, targetdir, objname)
+            else:
+                objpath = os.path.join(proj_dir_to_build_root, objname)
```